src/server/routers/update-logs.ts

```diff
@@ -0,0 +1,294 @@
+import { z } from 'zod';
```

Comment:
since the log file is essentially .ndjson, consider using `src/server/helpers/ndjson.ts`.